### PR TITLE
add support for installing more than one portal

### PR DESCRIPTION
**Pull request by hossamabdullah**
#### What this PR resolves:
installing more than one portal through prefab

**Version**: 
9.3.0

### CODE REVIEW

File: modules/web/PrefabPortal.py [hunk 1]
```diff
@@ -21,13 +20,15 @@ def configure(
...
             client_secret='',
             organization='',
-            redirect_address=''):
+            redirect_address='',
+            name='main'):
 
+        self.portal_name = name
         cfg = self.prefab.executor.state.configGet('portal')
-        cfg['main']['production'] = production
+        cfg[self.portal_name]['production'] = production
```
Comment (rkhamis) [inline modules/web/PrefabPortal.py]: why not `name` instead of `self.portal_name`?

File: modules/web/PrefabPortal.py [hunk 2]
```diff
@@ -21,13 +20,15 @@ def configure(
...
 
         if production:
-            oauth_cfg = cfg['main']['oauth']
+            oauth_cfg = cfg[self.portal_name]['oauth']
```
Comment (rkhamis) [inline modules/web/PrefabPortal.py]: same.

File: modules/web/PrefabPortal.py [hunk 3]
```diff
@@ -36,50 +37,39 @@ def configure(
             oauth_cfg['organization'] = organization
             oauth_cfg['redirect_url'] = 'http://%s/restmachine/system/oauth/authorize' % redirect_address
 
-        cfg['main']['mongoengine'] = {'host': mongodbip, 'port': mongoport}
+        cfg[self.portal_name]['mongoengine'] = {'host': mongodbip, 'port': mongoport}
```
Comment (rkhamis) [inline modules/web/PrefabPortal.py]: same

File: modules/web/PrefabPortal.py [hunk 4]
```diff
@@ -36,50 +37,39 @@ def configure(
...
-        self.prefab.core.file_write(self.configPath, j.data.serializer.yaml.dumps(cfg))
-
-    def add_configuration(self, config_dict):
-        """
-        use this method when ever u want to add some config to portal
-        usually when u add a new space
-        """
-        content = self.prefab.core.file_read('$TEMPLATEDIR/cfg/portal/config.yaml')
-        cfg = j.data.serializer.yaml.loads(content)
+        self.portal_name = name
```
Comment (rkhamis) [inline modules/web/PrefabPortal.py]: I'd prefer we directly use the name

File: modules/web/PrefabPortal.py [hunk 5]
```diff
@@ -36,50 +37,39 @@ def configure(
...
+        cfg = self.prefab.executor.state.configGet('portal')
         for key, value in config_dict.items():
-            cfg[key] = value
-        self.prefab.core.file_write(self.configPath, j.data.serializer.yaml.dumps(cfg))
-
+            cfg[self.portal_name][key] = value
+        self.prefab.executor.state.configSet('portal',cfg)
 
     @property
     def configPath(self):
```
Comment (rkhamis) [inline modules/web/PrefabPortal.py]: why do you even need this? What's the usecase of this property?

Comment (hossamabdullah) [reply modules/web/PrefabPortal.py]: It was already in the class. I was not sure if it has been deleted will something broke or no?

File: modules/web/PrefabPortal.py (hunk 4 above)
Comment (hossamabdullah) [reply modules/web/PrefabPortal.py]: the whole concept of using self.portal_name is not to break configPath property which was using the old main_dir attribute on the class

**Pull request merged**